### ISSUE TITLE
The device is not connected to the IoT hub

### ISSUE THREAD

**Issue by mr-suw** [TA2k/ioBroker.wmswebcontrol#7]
**Describe the bug**  
Start of adapter and login works, but than it runs into error. Log output is:

2023-07-08 18:14:19.750 - info: wmswebcontrol.0 (16255) starting. Version 0.0.5 in /opt/iobroker/node_modules/iobroker.wmswebcontrol, node: v18.16.1, js-controller: 4.0.23
--
2023-07-08 18:14:22.618 - info: wmswebcontrol.0 (16255) Login successful
2023-07-08 18:14:23.368 - error: wmswebcontrol.0 (16255) https://devicecloudservice.prod.devicecloud.warema.de/api/v1.0/communication/wcp/01394626/postMessage/
2023-07-08 18:14:23.369 - error: wmswebcontrol.0 (16255) Error: Request failed with status code 504
2023-07-08 18:14:23.371 - error: wmswebcontrol.0 (16255) {"code":4001,"message":"The device is not connected to the iot hub"}
2023-07-08 18:14:23.372 - error: wmswebcontrol.0 (16255) Get DevicesList failed
2023-07-08 18:14:23.374 - error: wmswebcontrol.0 (16255) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-07-08 18:14:23.374 - error: wmswebcontrol.0 (16255) unhandled promise rejection: Cannot read properties of undefined (reading 'response')
2023-07-08 18:14:23.379 - error: wmswebcontrol.0 (16255) TypeError: Cannot read properties of undefined (reading 'response')
at /opt/iobroker/node_modules/iobroker.wmswebcontrol/main.js:356:27
at processTicksAndRejections (node:internal/process/task_queues:95:5)
2023-07-08 18:14:23.379 - error: wmswebcontrol.0 (16255) Cannot read properties of undefined (reading 'response')
2023-07-08 18:14:23.405 - info: wmswebcontrol.0 (16255) terminating
2023-07-08 18:14:23.407 - warn: wmswebcontrol.0 (16255) Terminated (UNCAUGHT_EXCEPTION): Without reason
2023-07-08 18:14:23.457 - error: wmswebcontrol.0 (16255) https://devicecloudservice.prod.devicecloud.warema.de/api/v1.0/communication/wcp/01394626/postMessage/
2023-07-08 18:14:23.457 - error: wmswebcontrol.0 (16255) Error: Request failed with status code 504
2023-07-08 18:15:01.642 - info: wmswebcontrol.0 (16275) starting. Version 0.0.5 in /opt/iobroker/node_modules/iobroker.wmswebcontrol, node: v18.16.1, js-controller: 4.0.23
2023-07-08 18:15:04.092 - info: wmswebcontrol.0 (16275) Login successful
2023-07-08 18:15:04.464 - error: wmswebcontrol.0 (16275) https://devicecloudservice.prod.devicecloud.warema.de/api/v1.0/communication/wcp/01394626/postMessage/
2023-07-08 18:15:04.465 - error: wmswebcontrol.0 (16275) Error: Request failed with status code 504
2023-07-08 18:15:04.717 - error: wmswebcontrol.0 (16275) https://devicecloudservice.prod.devicecloud.warema.de/api/v1.0/communication/wcp/01394626/postMessage/
2023-07-08 18:15:04.718 - error: wmswebcontrol.0 (16275) Error: Request failed with status code 504
2023-07-08 18:15:04.719 - error: wmswebcontrol.0 (16275) {"code":4001,"message":"The device is not connected to the iot hub"}
2023-07-08 18:15:04.719 - error: wmswebcontrol.0 (16275) Get DevicesList failed
2023-07-08 18:15:04.721 - error: wmswebcontrol.0 (16275) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-07-08 18:15:04.721 - error: wmswebcontrol.0 (16275) unhandled promise rejection: Cannot read properties of undefined (reading 'response')
2023-07-08 18:15:04.725 - error: wmswebcontrol.0 (16275) TypeError: Cannot read properties of undefined (reading 'response')
at /opt/iobroker/node_modules/iobroker.wmswebcontrol/main.js:356:27
at processTicksAndRejections (node:internal/process/task_queues:95:5)
2023-07-08 18:15:04.726 - error: wmswebcontrol.0 (16275) Cannot read properties of undefined (reading 'response')
2023-07-08 18:15:04.752 - info: wmswebcontrol.0 (16275) terminating
2023-07-08 18:15:04.754 - warn: wmswebcontrol.0 (16275) Terminated (UNCAUGHT_EXCEPTION): Without reason
2023-07-08 18:15:43.078 - info: wmswebcontrol.0 (16296) starting. Version 0.0.5 in /opt/iobroker/node_modules/iobroker.wmswebcontrol, node: v18.16.1, js-controller: 4.0.23
2023-07-08 18:15:46.462 - info: wmswebcontrol.0 (16296) Login successful
2023-07-08 18:15:46.809 - error: wmswebcontrol.0 (16296) https://devicecloudservice.prod.devicecloud.warema.de/api/v1.0/communication/wcp/01394626/postMessage/
2023-07-08 18:15:46.809 - error: wmswebcontrol.0 (16296) Error: Request failed with status code 504
2023-07-08 18:15:46.810 - error: wmswebcontrol.0 (16296) {"code":4001,"message":"The device is not connected to the iot hub"}
2023-07-08 18:15:46.811 - error: wmswebcontrol.0 (16296) Get DevicesList failed
2023-07-08 18:15:46.812 - error: wmswebcontrol.0 (16296) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-07-08 18:15:46.812 - error: wmswebcontrol.0 (16296) unhandled promise rejection: Cannot read properties of undefined (reading 'response')
2023-07-08 18:15:46.817 - error: wmswebcontrol.0 (16296) TypeError: Cannot read properties of undefined (reading 'response')
at /opt/iobroker/node_modules/iobroker.wmswebcontrol/main.js:356:27
at processTicksAndRejections (node:internal/process/task_queues:95:5)
2023-07-08 18:15:46.818 - error: wmswebcontrol.0 (16296) Cannot read properties of undefined (reading 'response')
2023-07-08 18:15:46.846 - info: wmswebcontrol.0 (16296) terminating
2023-07-08 18:15:46.848 - warn: wmswebcontrol.0 (16296) Terminated (UNCAUGHT_EXCEPTION): Without reason
2023-07-08 18:15:47.030 - error: wmswebcontrol.0 (16296) https://devicecloudservice.prod.devicecloud.warema.de/api/v1.0/communication/wcp/01394626/postMessage/
2023-07-08 18:15:47.030 - error: wmswebcontrol.0 (16296) Error: Request failed with status code 504
2023-07-08 18:22:04.335 - info: admin.0 (832) ==> Connected system.user.iobroker from ::ffff:192.168.20.10
2023-07-08 18:22:12.972 - info: admin.0 (832) <== Disconnect system.user.iobroker from ::ffff:192.168.20.10
2023-07-08 18:22:18.043 - info: wmswebcontrol.0 (16342) starting. Version 0.0.5 in /opt/iobroker/node_modules/iobroker.wmswebcontrol, node: v18.16.1, js-controller: 4.0.23
2023-07-08 18:22:20.480 - info: wmswebcontrol.0 (16342) Login successful
2023-07-08 18:22:20.885 - error: wmswebcontrol.0 (16342) https://devicecloudservice.prod.devicecloud.warema.de/api/v1.0/communication/wcp/01394626/postMessage/
2023-07-08 18:22:20.885 - error: wmswebcontrol.0 (16342) Error: Request failed with status code 504
2023-07-08 18:22:20.886 - error: wmswebcontrol.0 (16342) {"code":4001,"message":"The device is not connected to the iot hub"}
2023-07-08 18:22:20.887 - error: wmswebcontrol.0 (16342) Get DevicesList failed
2023-07-08 18:22:20.889 - error: wmswebcontrol.0 (16342) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2023-07-08 18:22:20.889 - error: wmswebcontrol.0 (16342) unhandled promise rejection: Cannot read properties of undefined (reading 'response')
2023-07-08 18:22:20.893 - error: wmswebcontrol.0 (16342) TypeError: Cannot read properties of undefined (reading 'response')
at /opt/iobroker/node_modules/iobroker.wmswebcontrol/main.js:356:27
at processTicksAndRejections (node:internal/process/task_queues:95:5)
2023-07-08 18:22:20.894 - error: wmswebcontrol.0 (16342) Cannot read properties of undefined (reading 'response')
2023-07-08 18:22:20.919 - info: wmswebcontrol.0 (16342) terminating
2023-07-08 18:22:20.921 - warn: wmswebcontrol.0 (16342) Terminated (UNCAUGHT_EXCEPTION): Without reason
2023-07-08 18:22:20.980 - error: wmswebcontrol.0 (16342) https://devicecloudservice.prod.devicecloud.warema.de/api/v1.0/communication/wcp/01394626/postMessage/
2023-07-08 18:22:20.980 - error: wmswebcontrol.0 (16342) Error: Request failed with status code 504
2023-07-08 18:22:59.329 - info: wmswebcontrol.0 (16360) starting. Version 0.0.5 in /opt/iobroker/node_modules/iobroker.wmswebcontrol, node: v18.16.1, js-controller: 4.0.23
2023-07-08 18:23:02.476 - info: wmswebcontrol.0 (16360) Login successful
2023-07-08 18:23:02.892 - info: wmswebcontrol.0 (16360) State value to set for "wmswebcontrol.0.devices.01394626.serialNumber" has to be type "string" but received type "number"

**Comment by TA2k** [comment]
Please install from github and try again

**Comment by mr-suw** [comment]
Thanks for your first look. I reinstalled from GitHub and switched adapter to debug mode (some login details has been removed). The new log output is as follows:


2023-07-08 21:16:27.075 - info: wmswebcontrol.0 (17725) starting. Version 0.0.6 (non-npm: TA2k/ioBroker.wmswebcontrol) in /opt/iobroker/node_modules/iobroker.wmswebcontrol, node: v18.16.1, js-controller: 4.0.23
--
2023-07-08 21:16:27.503 - silly: wmswebcontrol.0 (17725) States system redis pmessage system.adapter.wmswebcontrol.0.logLevel/system.adapter.wmswebcontrol.0.logLevel:{"val":"silly","ack":true,"ts":1688843787482,"q":0,"from":"system.adapter.wmswebcontrol.0","lc":1688843787482}
2023-07-08 21:16:27.521 - silly: wmswebcontrol.0 (17725) States user redis pmessage wmswebcontrol.0.*/wmswebcontrol.0.info.connection:{"val":false,"ack":true,"ts":1688843787518,"q":0,"from":"system.adapter.wmswebcontrol.0","user":"system.user.admin","lc":1688843775778}
2023-07-08 21:16:30.280 - debug: wmswebcontrol.0 (17725) code: 76C5A121D50A8EACCF08554006CAB82BF9862EEAD760EC101814E593EDA05C4E-1
2023-07-08 21:16:30.281 - debug: wmswebcontrol.0 (17725) code_verifier: S651xd07nbfDuSSxYZMKmtPL7ivnW0WMDozYFIO6MwOIZTvQxuBZ26aPYkfsgdqK
2023-07-08 21:16:30.282 - debug: wmswebcontrol.0 (17725) codeChallenge: 36d7twLH80B4qpbXOZIWEBOUZJfZ4DrJDaSwUNjJFXw
2023-07-08 21:16:30.282 - debug: wmswebcontrol.0 (17725) state: 237AN7HkMZgRY1D9OrfjwBlSd3aI2RRHLTfGNdefyTqDQlA93m
2023-07-08 21:16:30.283 - debug: wmswebcontrol.0 (17725) nonce: 234qjK8VDgU3bXWAZEEUxPKgP0Tt7s4NBLiltU9kKmDjFwYT4t
2023-07-08 21:16:31.570 - info: wmswebcontrol.0 (17725) Login successful
2023-07-08 21:16:31.573 - debug: wmswebcontrol.0 (17725) get devices
2023-07-08 21:16:31.630 - silly: wmswebcontrol.0 (17725) States user redis pmessage wmswebcontrol.0.*/wmswebcontrol.0.info.connection:{"val":true,"ack":true,"ts":1688843791627,"q":0,"from":"system.adapter.wmswebcontrol.0","user":"system.user.admin","lc":1688843791627}
2023-07-08 21:16:31.793 - debug: wmswebcontrol.0 (17725) {"result":[{"articleNumber":"2020920","serialNumber":"01395627","type":"Wcp","registrationDate":"2022-07-08T15:09:24.9417289"}],"totalItemCount":1}
2023-07-08 21:16:31.808 - debug: wmswebcontrol.0 (17725) https://devicecloudservice.prod.devicecloud.warema.de/api/v1.0/communication/wcp/01394626/postMessage/
2023-07-08 21:16:31.809 - debug: wmswebcontrol.0 (17725) {"action":"mb8Read","parameters":{"address":0,"block":42,"eui":1394626,"length":12800},"changeIds":[]}
2023-07-08 21:16:31.837 - silly: wmswebcontrol.0 (17725) States user redis pmessage wmswebcontrol.0.*/wmswebcontrol.0.devices.01.articleNumber:{"val":"2020920","ack":true,"ts":1688843791833,"q":0,"from":"system.adapter.wmswebcontrol.0","user":"system.user.admin","lc":1688843602395}
2023-07-08 21:16:31.839 - silly: wmswebcontrol.0 (17725) States user redis pmessage wmswebcontrol.0.*/wmswebcontrol.0.devices.01.serialNumber:{"val":"01394626","ack":true,"ts":1688843791834,"q":0,"from":"system.adapter.wmswebcontrol.0","user":"system.user.admin","lc":1688843602396}
2023-07-08 21:16:31.841 - silly: wmswebcontrol.0 (17725) States user redis pmessage wmswebcontrol.0.*/wmswebcontrol.0.devices.01.type:{"val":"Wcp","ack":true,"ts":1688843791838,"q":0,"from":"system.adapter.wmswebcontrol.0","user":"system.user.admin","lc":1688843602397}
2023-07-08 21:16:31.918 - silly: wmswebcontrol.0 (17725) States user redis pmessage wmswebcontrol.0.*/wmswebcontrol.0.devices.01.registrationDate:{"val":"2021-07-08T15:09:24.9417289","ack":true,"ts":1688843791915,"q":0,"from":"system.adapter.wmswebcontrol.0","user":"system.user.admin","lc":1688843602475}
2023-07-08 21:16:32.123 - error: wmswebcontrol.0 (17725) https://devicecloudservice.prod.devicecloud.warema.de/api/v1.0/communication/wcp/01394626/postMessage/
2023-07-08 21:16:32.123 - error: wmswebcontrol.0 (17725) AxiosError: Request failed with status code 504
2023-07-08 21:16:32.124 - error: wmswebcontrol.0 (17725) {"code":4001,"message":"The device is not connected to the iot hub"}
2023-07-08 21:16:32.125 - error: wmswebcontrol.0 (17725) Get DevicesList failed
2023-07-08 21:16:32.125 - error: wmswebcontrol.0 (17725) TypeError: Cannot read properties of undefined (reading 'response')
2023-07-08 21:16:32.130 - error: wmswebcontrol.0 (17725) https://devicecloudservice.prod.devicecloud.warema.de/api/v1.0/communication/wcp/01394626/postMessage/
2023-07-08 21:16:32.131 - error: wmswebcontrol.0 (17725) AxiosError: Request failed with status code 504
